### PR TITLE
Extract `GenerateTimeoutCode` function to reduce duplicate code, avoid magic numbers for generating timeout code

### PR DESCRIPTION
**Pull request by Robyt3**
## Checklist

- [X] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/gameclient.cpp [hunk 1]
```diff
@@ -202,6 +202,20 @@ void CGameClient::OnConsoleInit()
 	Console()->Chain("cl_menu_map", ConchainMenuMap, this);
 }
 
+static void GenerateTimeoutCode(char *pTimeoutCode)
+{
+	if(pTimeoutCode[0] == '\0' || str_comp(pTimeoutCode, "hGuEYnfxicsXGwFq") == 0)
```
Comment (furo321) [inline src/game/client/gameclient.cpp:207]: Where does this `hGuEYnfxicsXGwFq` come from and what does it solve?

**Review by def- — APPROVED** [review]

File: src/game/client/gameclient.cpp (hunk 1 above)
Comment (Robyt3) [reply src/game/client/gameclient.cpp:207]: It comes from f42b9d69046fccf1307f2268f64f0add6765fd17 but I don't know what this solved.

File: src/game/client/gameclient.cpp [hunk 2]
```diff
@@ -202,6 +202,20 @@ void CGameClient::OnConsoleInit()
...
+	{
+		for(unsigned int i = 0; i < 16; i++)
+		{
+			if(rand() % 2)
+				pTimeoutCode[i] = (char)((rand() % ('z' - 'a' + 1)) + 'a');
+			else
+				pTimeoutCode[i] = (char)((rand() % ('Z' - 'A' + 1)) + 'A');
```
Comment (heinrich5991) [inline src/game/client/gameclient.cpp:214]: `rand` is not suitable for secure generation. We should use `generate_password` instead.

Comment (Robyt3) [reply src/game/client/gameclient.cpp:214]: We already seem to generate secure random timeout codes with the seeded mechanism, maybe we should remove the legacy method to manually set a timeout code.

CC https://github.com/ddnet/ddnet/issues/8110#issuecomment-2001963224

File: src/game/client/gameclient.cpp (hunk 1 above)
Comment (Robyt3) [reply src/game/client/gameclient.cpp:207]: I guess the `rand` function was previously seeded with a constant, e.g. `0`, so every player had the same timeout code `hGuEYnfxicsXGwFq` by default.

File: src/game/client/gameclient.cpp (hunk 2 above)
Comment (heinrich5991) [reply src/game/client/gameclient.cpp:214]: Yea, sounds good.

**Pull request merged**